DOC-875: Adding information for causes and solutions of errors (#2085)
tinymce/tinymce-docs: cloud-deployment-guide/cloud-troubleshooting.md

---
layout: default
title: Cloud Troubleshooting
description_short: Troubleshooting errors shown by the Tiny Cloud
description: Causes and solutions to common issues when using Tiny Cloud
keywords: tinymce cloud script textarea apiKey troubleshooting banners domain referer
---

When {{site.cloudname}} detects a problem, it will show an editor notification containing information about the problem. This page describes the cause and solution for common {{site.cloudname}} error notifications.

> **NOTE**: The wording of the notifications shown here may differ from the actual notifications from {{site.cloudname}}.

## "This domain is not registered with Tiny Cloud. Please see the quick start guide or create an account."

### Cause

This notification is **only shown** when either:

- An `apiKey` is not provided in the script tag.

> - `no-api-key` is provided as the apiKey.

Reviewer: ```suggestion
- `no-api-key` is provided as the `apiKey`.
```

Reviewer: or even
```suggestion
- `no-api-key` is provided as the API key.
```

Such as:

```html
<script src="{{ site.cdnurl }}" referrerpolicy="origin"></script>
```

### Solution

> Update the `src` URL include your (website or application developer's) {{site.cloudname}} API Key. Your API key should replace the string `no-api-key`. For example, if your API is `abcd1234`:

Reviewer: ```suggestion
Update the `src` URL to include your (website or application developer's) {{site.cloudname}} API Key. Your API key should replace the string `no-api-key`. For example, if your API is `abcd1234`:
```

```html
<script src="https://cdn.tiny.cloud/1/abcd1234/tinymce/5/tinymce.min.js" referrerpolicy="origin"></script>
```

To retrieve your API key, or to sign up for an API key, visit: [{{site.cloudname}}]({{site.accountsignup}}).

## "The API key you have entered is invalid. Please review your API key."

### Cause

This notification is shown when the API key provided cannot be found on the {{site.cloudname}} server.

The `apiKey` must be:

- An exact length,
- comprised of certain characters, and
- created with a {{site.cloudname}} account on the [{{site.accountpage}} page]({{site.accountsignup}}).

### Solution

Check the `apiKey` provided in the script tag:

> - Remove any leading or trailing spaces

Reviewer: ```suggestion
- Remove any leading or trailing spaces.
```
For consistency

> - Any other characters that are not in your apiKey. If you are using variable substitution, ensure that the variable is substituting properly.

Reviewer: ```suggestion
- Any other characters that are not in your `apiKey`. If you are using variable substitution, ensure that the variable is substituting properly.
```

Reviewer: or again as I'm not sure which way this should go
```suggestion
- Any other characters that are not in your API key. If you are using variable substitution, ensure that the variable is substituting properly.
```

1. Open your browser's _Developer's Tools_.
1. Open the _Network_ tab.
1. Find and select the request being made to load {{site.productname}} from {{site.cloudname}} with your API key.
1. Click on the **Headers** tab.

In the section called **Request Headers** there should be a field for **Referer**. This is the value that {{site.productname}} is checking against your approved domains. It must match one of your approved domains listed on your [{{site.accountpage}}]({{site.accountpageurl}}).

### Solution

If the `Referer` is correct for the site, ensure that the domain is included in your list of approved domains on [{{site.accountpage}}]({{site.accountpageurl}}). If the `Referer` is not what you are expecting, you may need to adjust your application's [**Referer** header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer) settings.

## "We’re unable to check your domain because the referer header is missing. Please read the Guide on how to ensure your referer header is present, so we can then customize your editor experience."

### Cause

> This notification is shown if the [**Referer** header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer) is absent for the network request for loading {{site.productname}} from {{site.cloudname}}. {{site.cloudname}} verifies the domain {{site.productname}} is loading from by checking the domain of the **Referer** header in the network request.

Reviewer: ```suggestion
This notification is shown if the [**Referer** header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer) is absent for the network request when loading {{site.productname}} from {{site.cloudname}}. {{site.cloudname}} verifies the domain {{site.productname}} is loading from by checking the domain of the **Referer** header in the network request.
```

*Referer* headers are sometimes removed by browser settings or browser extensions. {{site.cloudname}} only needs the domain in the **Referer** header, so for improved performance and privacy {{site.companyname}} recommends setting the `referrerpolicy` to `origin` when requesting {{site.cloudname}} resources.

### Solution

Identify which browser setting or extension is responsible for blocking the **Referer** being sent. A common extension is `Referer Control`:

- [Google Chrome](https://chrome.google.com/webstore/detail/referer-control/hnkcfpcejkafcihlgbojoidoihckciin?hl=en)
- [Firefox](https://addons.mozilla.org/en-US/firefox/addon/referercontrol/)

Once you have identified the setting or extension, modify it to allow just the `Origin` of the `Referer` to be sent. Alternatively, disable the extension or setting for any pages using {{site.cloudname}}.

## "The ___ premium plugin is not enabled on your API key. Upgrade your account."

### Cause

> This notification is shown when your apiKey does not have access to the premium plugin being requested. This could be the result of a trial expiring, and your {{site.productname}} configuration attempting to load premium plugins you can no longer access.

Reviewer: ```suggestion
This notification is shown when your API key does not have access to the premium plugin being requested. This could be the result of a trial expiring, and your {{site.productname}} configuration attempting to load premium plugins you can no longer access.
```
I'm assuming at this point this is the right version to be using.

> You may also be seeing this notification if you are using the wrong apiKey. Ensure that you are using the apiKey that has the right entitlements.

Reviewer: ```suggestion
You may also be seeing this notification if you are using the wrong API key. Ensure that you are using the API key that has the right entitlements.
```

---

When attemting to load the editor, many possible problems can occur. These are mostly related to a incorrect combination of 
your `apiKey` and the page `referer`. When {{site.cloudname}} detects a problem, it will show an editor
notification containing information about the problem. You can find explanations for the causes and solutions to 
the {{site.cloudname}} notifications here.

> NOTE: The wording of these notifications might differ slightly from what {{site.cloudname}} is showing.

## Editor Notifications

### "This domain is not registered with Tiny Cloud. Please see the quick start guide or create an account."

#### Cause

This notification is **only shown** when you have either:

1. not provided an `apiKey` in the script tag.
1. used `no-api-key` as your apiKey

#### Solution

If you have already signed up for {{site.cloudname}}, then you need to ensure that your {{site.cloudname}} script source uses that apiKey. It should **not** include `no-api-key`.

If you have signed up for {{site.cloudname}}, you will need to signup to receive an apiKey.

### "The API key you have entered is invalid. Please review your API key."

#### Cause

> This notification is shown when you have provided an `apiKey`, but that `apiKey` has the wrong structure. Your `apiKey` must

Author: - [ ] I'm pretty sure this is wrong actually. I should double-check it. I think it will give this error if the apiKey can't be found as well.

Author: Yeah, it's wrong. Need to fix.

Reviewer 2: ```
### Cause

This notification is shown when either:

- The API key provided cannot be found on the {{site.cloudname}} server.
- The provided `apiKey` has the wrong structure.
```
?

Author: I'm not `100%` sure on the structure, actually .. but the "cannot be found" will true for an invalid instructure anyway, so maybe just say "cannot be found"

## "The API key you have entered is invalid. Please review your API key."

### Cause

This notification is shown when the API key provided cannot be found on the {{site.cloudname}} server.

The `apiKey` must be:

- An exact length,
- comprised of certain characters, and
- created with a {{site.cloudname}} account on the [{{site.accountpage}} page]({{site.accountsignup}}).

### Solution

Check the `apiKey` provided in the script tag:

- Remove any leading or trailing spaces
- Any other characters that are not in your apiKey. If you are using variable substitution, ensure that the variable is substituting properly.
- Matches the API key shown at {{site.accountpageurl}}.

## "This domain is not registered with Tiny Cloud. Please review your approved domains."

### Cause

{{site.cloudname}} verifies the domain {{site.productname}} is loading from by checking the domain portion of the **Referer** header in the network request. This notification is shown when the **Referer** of your page does not match the list of approved domains stored against your `apiKey`. You can
view what your approved domains are in [Account]({{site.accountpageurl}}).

> Sometimes the domains in the **Referer** header does not match with the URL in the browser's address bar. To check the **Referer** header, open your browser's _Developer's Tools_ and open the _Network_ tab. From there, find the request being made to load {{site.productname}} with your API key, and click on the **Headers** tab.  In the section called **Request Headers** there should be a field for **Referer**. This is the value that {{site.productname}} is checking against your approved domains. It must match one your approved domains, or you will receive this notification.

Author: ```suggestion
Sometimes the domains in the **Referer** header does not match with the URL in the browser's address bar. To check the **Referer** header, open your browser's _Developer's Tools_ and open the _Network_ tab. From there, find the request being made to load {{site.productname}} with your API key, and click on the **Headers** tab.  In the section called **Request Headers** there should be a field for **Referer**. This is the value that {{site.productname}} is checking against your approved domains. It must match one of your approved domains, or you will receive this notification.
```

### Solution

If the `Referer` is what you are expecting, then you need to ensure that domain is included in your list of approved domains. If the `Referer` is not what you are expecting, you may need to adjust your application's **Referer** header settings.

> ## "We could not check your domain because the referer header was missing. Please see the guide on how to ensure the referer header is present."

Author: ```suggestion
## "We’re unable to check your domain because the referer header is missing. Please read the Guide on how to ensure your referer header is present, so we can then customize your editor experience."
```

### Cause

{{site.cloudname}} verifies the domain {{site.productname}} is loading from by checking the domain portion of the **Referer** header in the network request. If the **Referer** header is absent, you will receive this notification.

*Referer* headers are sometimes stripped out by certain browser settings and browser extensions. Remember, {{site.cloudname}} only cares about the domain portion of the **Referer** header, so for improved performance and privacy we recommend setting the `referrerpolicy` to `origin` when requesting {{site.cloudname}} resources.

### Solution

Identify which setting or extension is responsible for blocking the **Referer** being sent. Some common extensions are:

* `Referer Control` - [Google Chrome](https://chrome.google.com/webstore/detail/referer-control/hnkcfpcejkafcihlgbojoidoihckciin?hl=en) and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/referercontrol/)

> *

Author: ```suggestion

```
